src/base/QXmppDataForm.h

```diff
@@ -120,10 +124,50 @@ class QXMPP_EXPORT QXmppDataForm
         QVariant value() const;
         void setValue(const QVariant &value);
 
+        void parse(const QDomElement &element);
+        void toXml(QXmlStreamWriter *writer) const;
     private:
         QSharedDataPointer<QXmppDataFormFieldPrivate> d;
     };
 
+    class QXMPP_EXPORT Item
+    {
+    public:
+        Item(const QXmppDataForm::Item &other);
+        Item();
+        ~Item();
+
+        QXmppDataForm::Item& operator=(const QXmppDataForm::Item &other);
+
+        QList<Field> fields() const;
+        QList<Field> &fields();
+        void setFields(const QList<QXmppDataForm::Field> &fields);
+
+        void parse(const QDomElement &element);
+        void toXml(QXmlStreamWriter *writer) const;
+    private:
+        QSharedDataPointer<QXmppDataFormItemPrivate> d;
+    };
+
+    class QXMPP_EXPORT Reported
```

Comment:
yes, both of them described a list of fields, the difference between them is reported stand for table haed and item stand for talbe cell.

reference:
the definition
    http://xmpp.org/extensions/xep-0004.html#protocol-results
the example
    https://xmpp.org/extensions/xep-0055.html#example-9